src/gtk/spinctrl.cpp

```diff
@@ -364,27 +365,31 @@ wxSize wxSpinCtrlGTKBase::DoGetSizeFromTextSize(int xlen, int ylen) const
 {
     wxASSERT_MSG( m_widget, wxS("GetSizeFromTextSize called before creation") );
 
+    // This is a bit stupid as we typically compute xlen by measuring some
+    // string of digits in the first place, but there doesn't seem to be
+    // anything better to do (unless we add some GetSizeFromNumberOfDigits()).
+    const double widthDigit = GetTextExtent("0123456789").GetWidth() / 10.0;
+    const int numDigits = wxRound(xlen / widthDigit);
```

Comment:
Can we override GetSizeFromText and just strlen the input text?

Reply:
It's currently not virtual, but it could be indeed made virtual and overridden here. Is this really better than adding `GetSizeFromNumberOfDigits()`? Not sure... in any case, this is rather low priority IMO, because, as I wrote in the ticket, neither should be necessary anyhow as `wxSpinCtrl` is supposed to determine its best size correctly automatically.